src/map.ts

```diff
@@ -98,19 +97,24 @@ type StrictOut<T extends IterableContainer, K> = {
 };
 
 export namespace map {
+  /**
+   * @deprecated use `map` directly, `indexed` modifier no longer needed.
+   */
   export function indexed<T, K>(
     array: ReadonlyArray<T>,
     fn: PredIndexed<T, K>,
   ): Array<K>;
+
+  /**
+   * @deprecated use `map` directly, `indexed` modifier no longer needed.
+   */
   export function indexed<T, K>(
     fn: PredIndexed<T, K>,
   ): (array: ReadonlyArray<T>) => Array<K>;
+
   export function indexed(): unknown {
-    return purry(_map(true), arguments, map.lazyIndexed);
+    return purry(mapImplementation, arguments, _lazy);
   }
 
-  export const lazy = _lazy(false);
-  export const lazyIndexed = _toLazyIndexed(_lazy(true));
-
```

Comment:
`lazy` is used for #520 , but in a purported v2 we can rewrite to properly use iterators